dbos-rules.ts

```diff
@@ -0,0 +1,435 @@
+import { TypeChecker } from "typescript";
+import * as tslintPlugin from "@typescript-eslint/eslint-plugin";
+import { ESLintUtils, TSESLint, TSESTree, ParserServicesWithTypeInformation } from "@typescript-eslint/utils";
+
+import {
+  createWrappedNode, Node, FunctionDeclaration,
+  CallExpression, ConstructorDeclaration, ClassDeclaration,
+  MethodDeclaration
+} from "ts-morph";
+
+// Should I find TypeScript variants of these?
+const secPlugin = require("eslint-plugin-security");
+const noSecrets = require("eslint-plugin-no-secrets");
+
+type EslintNode = TSESTree.Node;
+type EslintContext = TSESLint.RuleContext<string, unknown[]>;
+
+//////////////////////////////////////////////////////////////////////////////////////////////////// Here is my `ts-morph` linting code:
+
+////////// These are some shared types and values used throughout the code
+
+// TODO: support `FunctionExpression` and `ArrowFunction` too
+type FunctionOrMethod = FunctionDeclaration | MethodDeclaration | ConstructorDeclaration;
+
+// This returns `undefined` if there is no error message to emit; otherwise, it returns a key to the `ERROR_MESSAGES` map
+type DetChecker = (node: Node, fn: FunctionOrMethod, isLocal: (name: string) => boolean) => string | undefined;
+
+type GlobalTools = {eslintContext: EslintContext, parserServices: ParserServicesWithTypeInformation, typeChecker: TypeChecker};
+let GLOBAL_TOOLS: GlobalTools | undefined = undefined;
+
+// These included `Transaction` and `TransactionContext` respectively before!
```

Comment:
What does this mean? I think we want to add Transaction and TransactionContext in the following lines.

Reply:
Yes - Peter told me we should leave them out; they were there before, but keeping them in resulted in lots of false positives on the `dbos-demo-apps` projects

Reply:
Yeah, we should only check workflows

Reply:
Okay, let's do it in a future PR.
I think your new rule that allows `await` on a function with `ctxt` as the first argument would already eliminate most of the false positives.